code/modules/mob/inventory.dm

```diff
@@ -508,3 +508,10 @@
 			SEND_SIGNAL(A, COMSIG_TRY_STORAGE_RETURN_INVENTORY, item_stuff)
 			processing_list += item_stuff
 	return processing_list
+
+//Checks carbon-based wearable item slots.
+/mob/proc/check_wearable_slot(mob/M, item, slot)
+	if(iscarbon(M))
+		if(istype(M.get_item_by_slot(slot), item))
+			return TRUE
+	return FALSE
```

Comment:
99% sure there's already a helper for this. This doesn't improve the code quality much regardless. I'd recommend going with ike's flag suggestion

Reply:
me and ike talked about it briefly like 2-3 weeks ago and the conclusion was this solution was alright. I really don't like the idea of a flag for such a specific thing.

Reply:
Having specific exceptions in fundamental interaction procs is pretty meh. If @ike709's okay with it though then this is fine for merge